File: JavaCode/CountSquareSumTriples.java
package JavaCode;

// Test cases
public class CountSquareSumTriples {
	public static void main(String[] args) {
		Solution obj = new Solution();
		// case 1
		int n1 = 5;
		System.out.println("Result of test case 1 : " + obj.countTriples(n1)); // Expected: 2

		// case 2
		int n2 = 10;
		System.out.println("Result of test case 2 : " + obj.countTriples(n2)); // Expected: 4
	}	
}

public class Solution {
    public int countTriples(int n) {
        int maxSq = n * n;
        boolean[] isSquare = new boolean[maxSq + 1];
        for (int c = 1; c <= n; c++) {
            isSquare[c * c] = true;
        }

        int count = 0;

        for (int a = 1; a <= n; a++) {
            int a2 = a * a;
            for (int b = 1; b <= n; b++) {
                int sum = a2 + b * b;
                if (sum <= maxSq && isSquare[sum]) {
                    count++;
                }
            }
        }

        return count;
    }
}

File: c/Count Square Sum Triples.c
#include <stdio.h>
#include <stdlib.h>

int countTriples(int n);

// Test cases
int main() {
	// case 1
	int n1 = 5;
	printf("Result of test case 1 : %d\n", countTriples(n1)); // Expected: 2
	
	// case 2
	int n2 = 10;
	printf("Result of test case 2 : %d\n", countTriples(n2)); // Expected: 4

	return 0;
}

int countTriples(int n) {
    int maxSq = n * n;
    int *sqToVal = (int*)calloc(maxSq + 1, sizeof(int));

    for (int c = 1; c <= n; c++) {
        sqToVal[c * c] = c;
    }

    int count = 0;

    for (int a = 1; a <= n; a++) {
        int a2 = a * a;
        for (int b = 1; b <= n; b++) {
            int s = a2 + b * b;
            if (s <= maxSq && sqToVal[s] != 0) {
                count++;
            }
        }
    }

    free(sqToVal);
    return count;
}

